sdk/tables/Azure.Data.Tables/src/TableClient.cs

```diff
@@ -471,16 +476,23 @@ public virtual Response<TableItem> CreateIfNotExists(CancellationToken cancellat
             scope.Start();
             try
             {
+                var context = new RequestContext { CancellationToken = cancellationToken, ErrorOptions = ErrorOptions.NoThrow };
```

Comment:
If CancellationToken is unset, you can potentially use a static and avoid an allocation here.

Reply:
This is typically only called once per client lifetime, so I think I'll leave it as is for now.